app/services/pii/cipher.rb

```diff
@@ -0,0 +1,39 @@
+module Pii
+  class Cipher
+    delegate :random_key, to: :cipher
+
+    def initialize
+      @cipher = OpenSSL::Cipher.new 'AES-256-CBC'
```

Comment:
I believe a new `Pii::Cipher` gets instantiated for each encrypt/decrypt, which means a new `OpenSSL::Cipher` is also instantiated. Did you find an example where that isn't true?